lib/datastore/query.js

```diff
@@ -109,18 +107,17 @@ Query.prototype.autoPaginate = function(autoPaginateVal) {
  */
 Query.prototype.filter = function(filter, value) {
   // TODO: Add filter validation.
-  var query = extend(true, new Query(), this);
   var operatorRegex = /[><=]/g;
   filter = filter.trim();
   var fieldName = filter.replace(operatorRegex, '').trim();
   var op = filter.substr(fieldName.length, filter.length).trim();
-  query.filters = query.filters || [];
-  query.filters.push({
+  this.filters = this.filters || [];
```

Comment:
These things are in the constructor it looks like. You can just push to them without `|| []`